pkg/recommendation/recommender/service/filter.go

```diff
@@ -0,0 +1,37 @@
+package service
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/gocrane/crane/pkg/recommendation/framework"
+)
+
+// Filter out k8s resources that are not supported by the recommender.
+func (s *ServiceRecommender) Filter(ctx *framework.RecommendationContext) error {
+	var err error
+
+	// filter resource that not match objectIdentity
+	if err = s.BaseRecommender.Filter(ctx); err != nil {
+		return err
+	}
+	var svc corev1.Service
+	if err = framework.ObjectConversion(ctx.Object, &svc); err != nil {
+		return err
+	}
+
+	if svc.Spec.Type != corev1.ServiceTypeLoadBalancer {
+		return fmt.Errorf("service: %v type: %s is not a LoadBalancer", ctx.Object.GetName(), svc.Spec.Type)
+	}
+
+	if err = framework.RetrievePods(ctx); err != nil {
```

Comment:
We also need to consider some condition like:
1. has no endpoints
2. the service and deployment are just created or updated

Reply:
Which stage would be more appropriate to implement these conditions？